src/Orleans.Runtime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -463,7 +479,7 @@ public async Task<AddressAndTag> RegisterAsync(ActivationAddress address, bool s
             // on all silos other than first, we insert a retry delay and recheck owner before forwarding
             if (hopCount > 0 && forwardAddress != null)
             {
-                await Task.Delay(RETRY_DELAY);
+                await this.RefreshMembership();
```

Comment:
Instead of an arbitrary wait, refresh membership once and wait for that refresh to propagate to the directory